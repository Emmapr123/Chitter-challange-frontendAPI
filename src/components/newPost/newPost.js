import React, { useState, useEffect } from 'react';
import { Button, ScrollView, StyleSheet, Text, TextInput, View, Dimensions, KeyboardAvoidingView } from 'react-native';
import { instance } from '../../axios/axios';

const NewPostComponent = (props) => {

  const [postBody, setPostBody] = useState('')

  const CreateNewPost = async() => {
    instance.post("/peeps", 
    {"peep": {"user_id": props.route.params.user_id, "body": postBody}}
    )
    .then(() => props.getPosts())
    .catch((error) => {
      console.log(error)
    })
  }

  return(
    <View>
      <TextInput
      placeholder="What is on your mind?"
      style={styles.textInput}
      onChangeText={(text) => setPostBody(text)}
      value={postBody}
      />
      <View style={styles.peepButton}>
        <Button title="Peep" onPress={CreateNewPost}/>
      </View>
    </View>
  )
}

const styles = StyleSheet.create({
  textInput: {
    paddingTop: 15,
    paddingBottom: 15,
    paddingLeft: 14,
    paddingRight: 14,
    fontSize: 16,
    color: 'black',
    backgroundColor: '#e3d0d8',
    width: '100%',
    alignItems: 'center',
  },
  peepButton: {
    alignItems: 'flex-end',
    color: '#e7e6f7',
    backgroundColor: '#e3d0d8',
  }
})

export { NewPostComponent }