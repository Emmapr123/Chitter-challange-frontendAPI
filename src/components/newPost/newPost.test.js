import 'react-native';
import React from 'react';

import { NewPostComponent } from './newPost.js';
import renderer from 'react-test-renderer';

test('it works', () => {
  expect(true).toBeTruthy();
});

describe("<NewPostComponent />", () => {
  const tree = renderer.create(<NewPostComponent />).toJSON();

  it('renders correctly', async () => {
    expect(tree).toMatchSnapshot();
  })

  it('has 2 children', async () => {
    expect(tree.children.length).toBe(2);
  });
});