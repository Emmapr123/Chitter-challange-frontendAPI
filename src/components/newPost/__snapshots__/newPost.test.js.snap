// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewPostComponent /> renders correctly 1`] = `
<View>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="What is on your mind?"
    rejectResponderTermination={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#e3d0d8",
        "color": "black",
        "fontSize": 16,
        "paddingBottom": 15,
        "paddingLeft": 14,
        "paddingRight": 14,
        "paddingTop": 15,
        "width": "100%",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "backgroundColor": "#e3d0d8",
        "color": "#e7e6f7",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Peep
        </Text>
      </View>
    </View>
  </View>
</View>
`;
